packages/gatsby-plugin-sharp/src/gatsby-node.js

```diff
@@ -78,15 +78,58 @@ exports.onCreateDevServer = async ({ app, cache, reporter }) => {
       return next()
     }
 
-    await _unstable_createJob(cacheResult, { reporter })
-    // we should implement cache.del inside our abstraction
-    await cache.cache.del(jobContentDigest)
+    // We are going to run a job for a single operation only
+    // and postpone all other operations
+    const {
+      matchingJob,
+      jobWithRemainingOperations,
+    } = splitOperationsByRequestedFile(cacheResult, pathOnDisk)
+
+    await _unstable_createJob(matchingJob, { reporter })
     await cache.cache.del(decodedURI)
 
+    if (jobWithRemainingOperations.args.operations.length > 0) {
+      // There are still some operations pending for this job - replace the cached job
+      await cache.cache.set(jobContentDigest, jobWithRemainingOperations)
+    } else {
+      // No operations left to process - purge the cache
+      await cache.cache.del(jobContentDigest)
+    }
+
     return res.sendFile(pathOnDisk)
   })
 }
 
+// Split the job into two jobs:
+//  - first job with a single operation matching requestedPathOnDisk
+//  - second job with all other operations
+// so the two resulting jobs are only different by their operations
+function splitOperationsByRequestedFile(job, requestedPathOnDisk) {
+  const matchingJob = {
+    ...job,
+    args: { ...job, operations: [] },
```

Comment:
I suppose this is job.args?
```suggestion
    args: { ...job.args, operations: [] },
```